### PR TITLE
Remove wandb.compat as unnecessary

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
Since we are now requiring `python >= 3.6`, `wandb.compat` can be removed.

Testing
-------
Ran the CI. Also need to check standalone/regression tests.

### CODE REVIEW

**Comment by dmitryduev** [comment]
@annirudh wanted to double check with you if it was ok to rm [this](https://github.com/wandb/client/pull/3192/files#diff-f7cbdfa3b0ee82d7dc14281c79d121ef2862759bee13b4c69a5a7804d8b7a1a6L160) stuff in FilePusher since you were the last one to touch it, according to git blame :)

File: wandb/sdk/internal/file_pusher.py [hunk 1]
```diff
@@ -79,11 +60,6 @@ def __init__(self, api, file_stream, silent=False):
         )
         self._step_upload.start()
 
-        # Holds refs to tempfiles if users need to make a temporary file that
-        # stays around long enough for file pusher to sync
-        # TODO(artifacts): maybe don't do this
-        self._temp_file_refs = []
```
Comment (dmitryduev) [inline wandb/sdk/internal/file_pusher.py:85]: I don't think this is used anywhere.

File: wandb/sdk/internal/file_pusher.py [hunk 2]
```diff
@@ -157,14 +133,6 @@ def store_manifest_files(self, manifest, artifact_id, save_fn):
...
         self._incoming_queue.put(event)
 
-    def named_temp_file(self, mode="w+b"):
-        # get a named temp file that the file pusher with hold a reference to so it
-        # doesn't get gc'd. Obviously, we shouldn't do this very much :). It's currently
-        # used for artifact metadata.
-        f = builtin_tempfile.NamedTemporaryFile(mode=mode, delete=False)
-        self._temp_file_refs.append(f)
-        return f
-
```
Comment (dmitryduev) [inline wandb/sdk/internal/file_pusher.py:167]: I don't think this method is used anywhere.

Comment (vwrj) [reply wandb/sdk/internal/file_pusher.py:167]: looks like it!

**Review by vwrj — APPROVED** [review]
LGTM!

**Pull request merged**